src/kixi/search/elasticsearch.clj

```diff
@@ -310,8 +310,32 @@
       (error e)
       (throw e))))
 
-(defn create-index
+(defn index-exists?
+  [es-url index-name]
+  (->> (str es-url "/" index-name)
+       client/head
+       :status
+       (= 200)))
+
+(defn create-index-
   [es-url index-name definition]
   (client/put (str es-url "/" index-name)
               {:body (json/generate-string definition)
                :headers {:content-type "application/json"}}))
+
+(defn create-index
+  [es-url index-name doc-type definition]
+  (create-index- es-url
+                 index-name
+                 {:mappings {doc-type
+                             {:properties (all-keys->es-format definition)}}
+                  :settings {:number_of_shards 1 ;; See https://www.elastic.co/guide/en/elasticsearch/guide/master/relevance-is-broken.html
```

Comment:
Are we sure we don't want to actually distribute documents?  Did you actually see issues in testing?

Relevant bits from the docs you linked:

"In practice, this is not a problem. The differences between local and global IDF diminish the more documents that you add to the index. With real-world volumes of data, the local IDFs soon even out. The problem is not that relevance is broken but that there is too little data.

For testing purposes, there are two ways we can work around this issue. The first is to create an index with one primary shard, as we did in the section introducing the match query. If you have only one shard, then the local IDF is the global IDF.

The second workaround is to add ?search_type=dfs_query_then_fetch to your search requests. The dfs stands for Distributed Frequency Search, and it tells Elasticsearch to first retrieve the local IDF from each shard in order to calculate the global IDF across the whole index."

Reply:
This is not for prod, just messing about.